## Union Probabilities

$$
P(A \text{ AND } B) = P(A) \text{ x } P(B)
$$
$$
P(A \text{ OR } B) = P(A) + P(B) - P(A \text{ AND } B)
$$
$$
P(A \text{ OR } B) = P(A) + P(B) - P(A) \text{ x } P(B)
$$

## Conditional Probabilities and Bayes's Theorem

expressed as $P(A \text{ GIVEN } B) \text{ or } P(A|B)$

$$
P(A|B) = \frac {P(B|A)\text{x}P(A)}{P(B)}
$$

\begin{split}
P(Coffee|Caner) = 0.85\\
P(Coffee) = 0.65\\
P(Cancer) = 0.005\\
\end{split}

what is the prob of someone haing canver given they are a coffee dinker ?

$P(Cancer|Coffee)$ ??

Using Baye's rule...

$$
P(Cancer|Coffee) = \frac {P(Coffee|Cancer)*P(Coffee)}{P(Cancer)}
$$

$$
P(Cancer|Coffee) = \frac {0.85 * 0.005}{0.65} = 0.0065
$$





## Binomial Distribution
measures how likely *k* successes can happen out of *n* trials given *p* probability.

Let’s say you are working on a new turbine jet engine and you ran 10 tests. The
outcomes yielded eight successes and two failures:

✓ ✓ ✓ ✓ ✓ ✘ ✓ ✘ ✓ ✓

You were hoping to get a 90% success rate, but based on this data you conclude
that your tests have failed with only 80% success. Each test is time-consuming and
expensive, so you decide it is time to go back to the drawing board to reengineer the design.
However, one of your engineers insists there should be more tests. “The only way we
will know for sure is to run more tests,” she argues. “What if more tests yield 90% or greater success? After all, if you flip a coin 10 times and get 8 heads, it does not mean the coin is fixed at 80%.”

![Binomial Distribution](images/Probabilities_Binomial.png)

Above, we see the probability of k successes for each bar out of 10 trials. This binomial distribution assumes a probability p of 90%, meaning there is a .90 (or 90%) chance for a success to occur. If this is true, that means there is a .1937 probability we would get 8 successes out of 10 trials. The probability of getting 1 success out of 10 trials is extremely unlikely, .000000008999, hence why the bar is not even visible. We can also calculate the probability of eight or fewer successes by adding up bars for eight or fewer successes. This would give us .2639 probability of eight or fewer successes.

In [4]:
from scipy.stats import binom

n = 10
p = 0.9

for k in range(n + 1):
    probability = binom.pmf(k, n, p)
    print("{0} - {1}".format(k, probability))

0 - 9.999999999999973e-11
1 - 8.999999999999976e-09
2 - 3.6449999999999933e-07
3 - 8.747999999999988e-06
4 - 0.00013778099999999974
5 - 0.0014880347999999982
6 - 0.011160260999999989
7 - 0.05739562799999997
8 - 0.1937102444999998
9 - 0.38742048899999976
10 - 0.34867844010000015


## Beta Distribution
The beta distribution allows us to see the likelihood of different underlying probabilities for an event to occur given alpha successes and beta failures.

![Beta Distribution](images/Probabilities_BetaDistribution.png)

The beta distribution is a type of probability distribution, which means the area under the entire curve is 1.0, or 100%. To find a probability, we need to find the area within a range. For example, if we want to evaluate the probability 8/10 successes would yield 90% or higher success rate, we need to find the area between 0.9 and 1.0, which is .225, as shaded

---

As we did with the binomial distribution, we can use SciPy to implement the beta distribution. Every continuous probability distribution has a cumulative density function (CDF), which calculates the area up to a given x-value. It is easy enough to use SciPy with its beta.cdf() function, and the only parameters we need to provide are the *x-value*, the number of successes *a*, and the number of failures
*b*

In [6]:
from scipy.stats import beta
a = 8
b = 2
p = beta.cdf(.90, a, b)

print(p)
# So according to our calculation, 
# there is a 77.48% chance the underlying probability
# of success is 90% or less

0.7748409780000002


In [8]:
# How do we calculate the probability of success being 90% or more
p = 1.0 - beta.cdf(.90, a, b)
print(p)

0.22515902199999982


This means that out of 8/10 successful engine tests, there is only a 22.5% chance the underlying success rate is 90% or greater. But there is about a 77.5% chance it is less than 90%. The odds are not in our favor here that our tests were successful, but we could gamble on that 22.5% chance with more tests if we are feeling lucky.
The beta distribution is a fascinating tool to measure the probability of an event
occurring versus not occurring, based on a limited set of observations. It allows us to reason about probabilities of probabilities, and we can update it as we get new data.
